src/native/libs/System.Security.Cryptography.Native/pal_evp_kem.c

```diff
@@ -0,0 +1,301 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "openssl.h"
+#include "pal_evp_kem.h"
+#include "pal_utilities.h"
+
+#include <assert.h>
+
+static int32_t GetKeyOctetStringParam(const EVP_PKEY* pKey,
```

Comment:
do we need this to be a separate subroutine? If there is a benefit we could share this code with https://github.com/dotnet/runtime/blob/main/src/native/libs/System.Security.Cryptography.Native/pal_ecc_import_export.c#L867 and all similar calls

Reply:
I'm not exactly sure what you are proposing, other than noting there are two things that use `EVP_PKEY_get_octet_string_param`.

In general I am a little wary of trying to do so much deduplication that it reaches the point of inscrutability. If there is something sensible we can do perhaps it can be done in a follow up PR.